### PR TITLE
feat: [#39] Implemented forgot password

### LINKED ISSUE #39: Implement forgot password
Make it with base form.

### PR DESCRIPTION
**Pull request by eldarcelik**
### What has changed and why?

- Created `RequestResetPassword ` and `ResetPassword ` forms and routes for them.
- Created `requestResetPassword` and `resetPassword` APIs.
- Added email and password validation patterns.
- Created `IAccountInfo` shared interface.
- Closes #39 

### Branch Name Validation

- [x] Branch name pattern: `{type}/{JIRA-ID}` or `{type}/issue-{GITHUB-ISSUE-ID}`

### PR Title Validation

- [x] PR title pattern: `{type}: [{JIRA-ID}] Description` or `{type}: [#{GITHUB-ISSUE-ID}] Description`

### Only check what is relevant for this PR

- [x] Styling verified for different screen sizes and browsers
- [x] Functionality verified for different screen sizes and browsers
- [ ] Relevant unit tests added/updated
- [ ] Storybook files added/updated

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import PrivateRoute from './components/PrivateRoute';
 import { MoviesShowsProvider } from './context';
 import Login from './pages/Account/Login';
 import Registration from './pages/Account/Registration';
+import RequestResetPassword from './pages/Account/RequestResetPassword';
+import ResetPassword from './pages/Account/ResetPassword/ResetPassword';
```
Comment (4zuko) [inline src/App.tsx:11]: Use index entry point.

File: src/pages/Account/RequestResetPassword/RequestResetPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    code: STATUS_CODES.OK,
+    message: '',
+  });
+
+  const handleEmail = (event: ChangeEvent<HTMLInputElement>): void => {
+    setEmailValidationMessage('');
+    setEmail(event.target.value);
+  };
+
+  const validateEmail = (email: string): boolean => {
```
Comment (4zuko) [inline src/pages/Account/RequestResetPassword/RequestResetPassword.tsx:24]: Should this custom validation be handled using yup? Or it's intentionally implemented?

File: src/pages/Account/RequestResetPassword/RequestResetPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      setEmailValidationMessage(MESSAGES.EMAIL_MUST_BE_VALID);
+
+      return false;
+    }
+
+    return true;
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
```
Comment (4zuko) [inline src/pages/Account/RequestResetPassword/RequestResetPassword.tsx:41]: Should we have a logic for `submitting` state? What will happen if the user clicks on the submit button multiple times?

File: src/pages/Account/types.ts [hunk 4]
```diff
@@ -8,3 +8,8 @@ export interface IUser {
 export interface ILogout {
   refreshToken: string;
 }
+
+export interface IAccountInfo {
```
Comment (4zuko) [inline src/pages/Account/types.ts:12]: Is there any sense to have the empty object as the account info?

**Review by 4zuko — APPROVED** [review]
Nice work. **`LGTM`**. I'll approve this PR but please check the comments.

File: src/pages/Account/RequestResetPassword/RequestResetPassword.tsx (hunk 2 above)
Comment (eldarcelik) [reply src/pages/Account/RequestResetPassword/RequestResetPassword.tsx:24]: Yes, it is intentionally implemented as we said that one form will be created without using any additional packages.

File: src/pages/Account/types.ts (hunk 4 above)
Comment (eldarcelik) [reply src/pages/Account/types.ts:12]: I changed that it can't be empty now.

**Pull request merged**